plugins/catalog-backend/src/ingestion/CatalogRules.ts

```diff
@@ -157,6 +172,9 @@ export class DefaultCatalogRulesEnforcer implements CatalogRulesEnforcer {
       if (matcher.target && matcher.target !== location?.target) {
         continue;
       }
+      if (matcher.match && !location?.target.match(matcher.match)) {
```

Comment:
I think we should avoid regex for this check, it's way too easy to write an insecure regex. In the [auth backend we've used minimatch](https://github.com/backstage/backstage/blob/4e12e8a0231123da294f475b0ffe4e98c3cb3cd8/plugins/auth-backend/src/service/router.ts#L206-L209), I'm thinking we could do that here too?